app/scripts/lib/metrics.js

```diff
@@ -484,10 +484,16 @@ define(function (require, exports, module) {
 
     logFlowEvent (eventName, viewName) {
       this.logEvent(marshallFlowEvent(eventName, viewName));
+      // We need to use server-side timestamps on flow events,
+      // so flush them to the server immediately.
+      this.flush();
```

Comment:
We can certainly go with the approach in this PR in the meantime, but I'm wondering what improvements can we make to make this even better...